src/online2/online-nnet2-feature-pipeline.cc

```diff
@@ -76,9 +84,14 @@ OnlineNnet2FeaturePipelineInfo::OnlineNnet2FeaturePipelineInfo(
   }
 }
 
+
+/// The main feature extraction pipeline is constructed in this constructor.
 OnlineNnet2FeaturePipeline::OnlineNnet2FeaturePipeline(
     const OnlineNnet2FeaturePipelineInfo &info):
-    info_(info) {
+    info_(info), base_feature_(NULL),
+    pitch_(NULL), pitch_feature_(NULL),
+    cmvn_feature_(NULL), ivector_feature_(NULL) {
+
   if (info_.feature_type == "mfcc") {
```

Comment:
maybe move most of these code to Init(), to keep consistency of its parent class OnlineFeaturePipeline 
 in src/online2/online-feature-pipeline.cc